<a href="https://colab.research.google.com/github/abrahamjenie/Programming-for-Data-Analysis-CA1-Web-Scraping/blob/main/SalaryData_WebScraping.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Python CA1 - WebScraping

Web scraping was done using the website https://www.salary.com, which contains salaries for various job roles in several countries. For this activity, data was extracted for different employment roles across various counties in Ireland

Loading necessary packages

In [1]:
#Importing Packages
import requests
from bs4 import BeautifulSoup
import json
import pandas as pd
import time

pd.set_option('display.max_colwidth', None) # This will display the entire content in each of the columns
pd.set_option('display.max_columns', None)  # This will display all the columns in the output

Checking the response for the website using the 'get' method

In [2]:
url="https://www.salary.com/research/ie-salary/alternate/data-scientist-analytics-intermediate-salary/ie/dublin"
r = requests.get(url)
print(r) #status code is 200 

<Response [200]>


In [ ]:
r.content


Using BeautifulSoup to parse the html content

In [4]:
soup=BeautifulSoup(r.content,'html.parser')
script=soup.find_all('script')[5].text
script

'\r\n    {\r\n    "@context": "http://schema.org",\r\n    "@type": "Occupation",\r\n    "name": "Data Scientist/Analytics , Intermediate",\r\n    "mainEntityOfPage": {\r\n    "@type": "WebPage",\r\n    "lastReviewed": "2021-03-19T00:00:00Z"\r\n    },\r\n    "description": "A Data Scientist is a professional who extensively works with Big Data in order to derive valuable business insights from it.\r\nHowever, data scientists and predictive analytics professionals (PAPs) are more likely to hold an advanced degree.\r\nDay-to-day activities of a Data Scientist sometimes can be predictable, and sometimes they are something out of the ordinary.\r\n",\r\n    "estimatedSalary": [{\r\n    "@type": "MonetaryAmountDistribution",\r\n    "name": "base",\r\n    "currency": "EUR",\r\n    "duration": "P1Y",\r\n    "percentile25": "31559",\r\n    "median": "39009",\r\n    "percentile75": "46472"\r\n    }]\r\n        \r\n            ,\r\n            "occupationLocation": [{\r\n            "@type": "City

As the above output is a dictionary, json library is used to extract further information

In [5]:
json_data=json.loads(script,strict=False) #strict=False is used to because there are control characters like '\r','\n'
json_data

{'@context': 'http://schema.org',
 '@type': 'Occupation',
 'name': 'Data Scientist/Analytics , Intermediate',
 'mainEntityOfPage': {'@type': 'WebPage',
  'lastReviewed': '2021-03-19T00:00:00Z'},
 'description': 'A Data Scientist is a professional who extensively works with Big Data in order to derive valuable business insights from it.\r\nHowever, data scientists and predictive analytics professionals (PAPs) are more likely to hold an advanced degree.\r\nDay-to-day activities of a Data Scientist sometimes can be predictable, and sometimes they are something out of the ordinary.\r\n',
 'estimatedSalary': [{'@type': 'MonetaryAmountDistribution',
   'name': 'base',
   'currency': 'EUR',
   'duration': 'P1Y',
   'percentile25': '31559',
   'median': '39009',
   'percentile75': '46472'}],
 'occupationLocation': [{'@type': 'City', 'name': 'Dublin, Ireland'}]}

Along with the above information, extracting url from the 'meta' html tag

In [6]:
meta_url=soup.find('meta',property="og:url")
meta_url

<meta content="https://www.salary.com/research/ie-salary/alternate/data-scientist-analytics-intermediate-salary/ie/dublin" property="og:url"/>

The details about the 'Data Scientist Intermediate' role in Dublin is displayed in the output below. It contains information such as the job title, median pay, 25th percentile salary, 75th percentile salary, job description, and so on.

In [7]:
job_designation=json_data['name']
job_desc=json_data['description']
percentile25_salary=json_data['estimatedSalary'][0]['percentile25']
median_salary=json_data['estimatedSalary'][0]['median']
percentile75_salary=json_data['estimatedSalary'][0]['percentile75']
salary_type=json_data['estimatedSalary'][0]['name']
currency=json_data['estimatedSalary'][0]['currency']
salary_duration=json_data['estimatedSalary'][0]['duration']
location=json_data['occupationLocation'][0]['name']
last_reviewed_date=json_data['mainEntityOfPage']['lastReviewed']
weblink=meta_url['content']

print(job_designation,percentile25_salary,median_salary,percentile75_salary,salary_type,currency,salary_duration,location,last_reviewed_date,weblink)

Data Scientist/Analytics , Intermediate 31559 39009 46472 base EUR P1Y Dublin, Ireland 2021-03-19T00:00:00Z https://www.salary.com/research/ie-salary/alternate/data-scientist-analytics-intermediate-salary/ie/dublin


Two ‘for’ loops are considered, one for iterating through cities and another through job titles. The only variation between the two URLs is the word alternate/benchmark. Some job titles use ‘alternate’ in the URL, while others use ‘benchmark’. An 'if-else' statement is used to determine whether the response status code is 200. The remaining part of the code is identical to the above code used to extract information for the 'Data Scientist Intermediate' job role.

In [8]:
cities=['dublin','cork','galway','limerick','shannon-west','waterford']
job_roles=['data-architect-ii','personal-assistant-to-ceo','operational-risk-analyst-ii','entry-hr-generalist','senior-risk-analyst','asset-and-liability-risk-analyst','junior-accountant','financial-accountant-i','it-user-support-manager','data-scientist-analytics-senior-specialist','life-sciences-research-scientist-iii','hr-assistant-ii','data-scientist-analytics-intermediate','entry-financial-analyst','entry-accountant','graphic-designer-ii','instructional-designer','healthcare-services-planning-director','warehouse-operations-supervisor','database-warehouse-programmer','warehouse-helper']
data={}
salary_df=pd.DataFrame(data)
for role in job_roles:
  for city in cities:
    alternate = 'https://www.salary.com/research/ie-salary/alternate/{}-salary/ie/{}'
    benchmark = 'https://www.salary.com/research/ie-salary/benchmark/{}-salary/ie/{}'

    url_alt = alternate.format(role, city)
    url_bench= benchmark.format(role, city)
    r_alt = requests.get(url_alt)
    r_bench=requests.get(url_bench)

    if r_alt.status_code==200 and r_bench.status_code!=200:
      r=requests.get(url_alt)
    elif r_bench.status_code==200 and r_alt.status_code!=200:
      r=requests.get(url_bench)
    else:
      print("HTTP request failed for {} and {}".format(url_alt,url_bench))
      continue

        
    soup=BeautifulSoup(r.content,'html.parser')
    script=soup.find_all('script')[5].text
    
    json_data=json.loads(script,strict=False)
    
    meta_url=soup.find('meta',property="og:url")
    
    new_row={'job_designation':json_data['name'],'percentile25_salary':json_data['estimatedSalary'][0]['percentile25'],
             'median_salary':json_data['estimatedSalary'][0]['median'],'percentile75_salary':json_data['estimatedSalary'][0]['percentile75'],
             'salary_type':json_data['estimatedSalary'][0]['name'],'currency':json_data['estimatedSalary'][0]['currency'],
             'salary_duration':json_data['estimatedSalary'][0]['duration'],'location':json_data['occupationLocation'][0]['name'],
             'last_reviewed_datetime':json_data['mainEntityOfPage']['lastReviewed'],'weblink':meta_url['content'],'job_desc':json_data['description']}
    salary_df = salary_df.append(new_row, ignore_index=True)
salary_df

job_designation percentile25_salary median_salary percentile75_salary  \
0    Data Architect II               42167         50115               58800   
1    Data Architect II               39972         47506               55739   
2    Data Architect II               39035         46393               54433   
3    Data Architect II               39420         46850               54969   
4    Data Architect II               38646         45930               53890   
..                 ...                 ...           ...                 ...   
121   Warehouse Helper               22562         29382               38055   
122   Warehouse Helper               22033         28693               37163   
123   Warehouse Helper               22251         28976               37530   
124   Warehouse Helper               21814         28407               36793   
125   Warehouse Helper               21406         27877               36106   

    salary_type currency salary_duration               location  \
0          base      EUR             P1Y        Dublin, Ireland   
1          base      EUR             P1Y          Cork, Ireland   
2          base      EUR             P1Y        Galway, Ireland   
3          base      EUR             P1Y      Limerick, Ireland   
4          base      EUR             P1Y  Shannon-West, Ireland   
..          ...      ...             ...                    ...   
121        base      EUR             P1Y          Cork, Ireland   
122        base      EUR             P1Y        Galway, Ireland   
123        base      EUR             P1Y      Limerick, Ireland   
124        base      EUR             P1Y  Shannon-West, Ireland   
125        base      EUR             P1Y     Waterford, Ireland   

    last_reviewed_datetime  \
0     2021-03-19T00:00:00Z   
1     2021-03-19T00:00:00Z   
2     2021-03-19T00:00:00Z   
3     2021-03-19T00:00:00Z   
4     2021-03-19T00:00:00Z   
..                     ...   
121   2021-03-19T00:00:00Z   
122   2021-03-19T00:00:00Z   
123   2021-03-19T00:00:00Z   
124   2021-03-19T00:00:00Z   
125   2021-03-19T00:00:00Z   

                                                                                          weblink  \
0          https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/dublin   
1            https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/cork   
2          https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/galway   
3        https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/limerick   
4    https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/shannon-west   
..                                                                                            ...   
121           https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/cork   
122         https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/galway   
123       https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/limerick   
124   https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/shannon-west   
125      https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/waterford   

                                                                                                                                                                                                                                                                                                                                                                              job_desc  
0                                                                          Contribute to an architecture roadmap covering transactional, operational and business intelligence.\r\nContribute to development of standards and practices for Java, JavaScript, and SQL development.\r\nContributes 

Checking for missing values and the dataypes of the features

In [9]:
#Checking for missing values in the dataset
salary_df.isna().sum()
#no missing values are observed

job_designation           0
percentile25_salary       0
median_salary             0
percentile75_salary       0
salary_type               0
currency                  0
salary_duration           0
location                  0
last_reviewed_datetime    0
weblink                   0
job_desc                  0
dtype: int64

In [10]:
#Checking the datatype of each feature
salary_df.info()
#variables like percentile25_salary, median_salary and percentile75_salary are treated as object instead of numeric

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 126 entries, 0 to 125
Data columns (total 11 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   job_designation         126 non-null    object
 1   percentile25_salary     126 non-null    object
 2   median_salary           126 non-null    object
 3   percentile75_salary     126 non-null    object
 4   salary_type             126 non-null    object
 5   currency                126 non-null    object
 6   salary_duration         126 non-null    object
 7   location                126 non-null    object
 8   last_reviewed_datetime  126 non-null    object
 9   weblink                 126 non-null    object
 10  job_desc                126 non-null    object
dtypes: object(11)
memory usage: 11.0+ KB


Data Cleaning


1.   Converting the datatypes for the features percentile25_salary, median_salary and percentile75_salary 
2.   Splitting the 'location' variable into two separate variables 'county' and 'country'
3.   Creating a new feature 'last_reviewed_date' that contains the datepart from 'last_reviewed_datetime'



In [11]:
#Data Cleaning

#Coverting the datatype from object to integer for the features percentile25_salary, median_salary  and salary_type 
salary_df=salary_df.astype({'percentile25_salary':'int','median_salary':'int','percentile75_salary':'int'})

#Splitting the 'location' feature into two new features 'county' and 'country'
location=salary_df['location'].str.split(",",expand=True) #expand=True will give the output in a dataframe instead of the default list
salary_df['county']=location[0]
salary_df['country']=location[1]

#Extracting only the date part from 'last_reviewed_datetime' and storing it to a new feature 'last_reviewed_date'
import datetime
salary_df['last_reviewed_date']=pd.to_datetime(salary_df['last_reviewed_datetime'].str.slice(0, 10))

salary_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 126 entries, 0 to 125
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   job_designation         126 non-null    object        
 1   percentile25_salary     126 non-null    int64         
 2   median_salary           126 non-null    int64         
 3   percentile75_salary     126 non-null    int64         
 4   salary_type             126 non-null    object        
 5   currency                126 non-null    object        
 6   salary_duration         126 non-null    object        
 7   location                126 non-null    object        
 8   last_reviewed_datetime  126 non-null    object        
 9   weblink                 126 non-null    object        
 10  job_desc                126 non-null    object        
 11  county                  126 non-null    object        
 12  country                 126 non-null    object    

Checking the summary statistics of the numeric features

In [12]:
salary_df.describe()

percentile25_salary  median_salary  percentile75_salary
count           126.000000     126.000000           126.000000
mean          39858.174603   49830.857143         61303.214286
std           13823.469212   18079.048392         22630.485614
min           21406.000000   27877.000000         36106.000000
25%           30484.000000   36273.750000         43665.250000
50%           37160.000000   45823.000000         56016.000000
75%           44992.750000   56217.250000         70726.250000
max           89038.000000  112863.000000        137388.000000

Checking the summary statistics of the numeric features for each job designation

In [13]:
salary_df.groupby(['job_designation']).describe()

percentile25_salary                \
                                                          count          mean   
job_designation                                                                 
Asset and Liability Risk Analyst                            6.0  35353.166667   
Data Architect II                                           6.0  39527.333333   
Data Scientist/Analytics , Intermediate                     6.0  29583.333333   
Data Scientist/Analytics, Senior Specialist                 6.0  65189.166667   
Database/Warehouse Programmer                               6.0  39239.666667   
Entry Accountant                                            6.0  28663.166667   
Entry Financial Analyst                                     6.0  32835.166667   
Entry HR Generalist                                         6.0  33769.166667   
Financial Accountant I                                      6.0  28739.000000   
Graphic Designer II                                         6.0  39539.333333   
HR Assistant II                                             6.0  28546.666667   
Healthcare Services Planning Director                       6.0  83464.500000   
IT User Support - Manager                                   6.0  46079.500000   
Instructional Designer                                      6.0  37892.833333   
Junior Accountant                                           6.0  28701.833333   
Life Sciences Research Scientist III                        6.0  54063.666667   
Operational Risk Analyst II                                 6.0  33349.333333   
Personal Assistant to CEO                                   6.0  46859.500000   
Senior Risk Analyst                                         6.0  45571.666667   
Warehouse Helper                                            6.0  22311.166667   
Warehouse Operations Supervisor                             6.0  37742.500000   

                                                                             \
                                                     std      min       25%   
job_designation                                                               
Asset and Liability Risk Analyst             1312.466444  33919.0  34652.00   
Data Architect II                            1467.440447  37924.0  38743.25   
Data Scientist/Analytics , Intermediate      1098.323207  28383.0  28996.75   
Data Scientist/Analytics, Senior Specialist  2419.788951  62545.0  63896.50   
Database/Warehouse Programmer                1456.676995  37648.0  38461.50   
Entry Accountant                             1063.938234  27501.0  28094.50   
Entry Financial Analyst                      1219.130742  31503.0  32183.75   
Entry HR Generalist                          1253.431995  32400.0  33099.25   
Financial Accountant I                       1066.699958  27574.0  28168.75   
Graphic Designer II                          1467.440447  37936.0  38755.25   
HR Assistant II                              1059.757457  27389.0  27980.25   
Healthcare Services Planning Director        3098.374332  80079.0  81809.25   
IT User Support - Manager                    1710.286263  44211.0  45165.50   
Instructional Designer                       1406.567654  36356.0  37141.25   
Junior Accountant                            1065.282013  27538.0  28132.50   
Life Sciences Research Scientist III         2007.054226  51871.0  52991.00   
Operational Risk Analyst II                  1237.774724  31997.0  32688.00   
Personal Assistant to CEO                    1739.573597  44959.0  45930.25   
Senior Risk Analyst                          1691.806687  43723.0  44668.00   
Warehouse Helper                              828.253443  21406.0  21868.75   
Warehouse Operations Supervisor              1401.110238  36212.0  36993.75   

                                                                         \
                                                 50%       75%      max   
job_designation                                     

The final dataset after pre-processing. There are 126 rows and 14 features.

In [14]:
#Final salary dataset after cleaning
salary_df

job_designation  percentile25_salary  median_salary  \
0    Data Architect II                42167          50115   
1    Data Architect II                39972          47506   
2    Data Architect II                39035          46393   
3    Data Architect II                39420          46850   
4    Data Architect II                38646          45930   
..                 ...                  ...            ...   
121   Warehouse Helper                22562          29382   
122   Warehouse Helper                22033          28693   
123   Warehouse Helper                22251          28976   
124   Warehouse Helper                21814          28407   
125   Warehouse Helper                21406          27877   

     percentile75_salary salary_type currency salary_duration  \
0                  58800        base      EUR             P1Y   
1                  55739        base      EUR             P1Y   
2                  54433        base      EUR             P1Y   
3                  54969        base      EUR             P1Y   
4                  53890        base      EUR             P1Y   
..                   ...         ...      ...             ...   
121                38055        base      EUR             P1Y   
122                37163        base      EUR             P1Y   
123                37530        base      EUR             P1Y   
124                36793        base      EUR             P1Y   
125                36106        base      EUR             P1Y   

                  location last_reviewed_datetime  \
0          Dublin, Ireland   2021-03-19T00:00:00Z   
1            Cork, Ireland   2021-03-19T00:00:00Z   
2          Galway, Ireland   2021-03-19T00:00:00Z   
3        Limerick, Ireland   2021-03-19T00:00:00Z   
4    Shannon-West, Ireland   2021-03-19T00:00:00Z   
..                     ...                    ...   
121          Cork, Ireland   2021-03-19T00:00:00Z   
122        Galway, Ireland   2021-03-19T00:00:00Z   
123      Limerick, Ireland   2021-03-19T00:00:00Z   
124  Shannon-West, Ireland   2021-03-19T00:00:00Z   
125     Waterford, Ireland   2021-03-19T00:00:00Z   

                                                                                          weblink  \
0          https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/dublin   
1            https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/cork   
2          https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/galway   
3        https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/limerick   
4    https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/shannon-west   
..                                                                                            ...   
121           https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/cork   
122         https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/galway   
123       https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/limerick   
124   https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/shannon-west   
125      https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/waterford   

                                                                                                                                                                                                                                                                                                                                                                              job_desc  \
0                                                                          Contribute to an architecture roadmap covering transactional, operational and business intelligence.\r\nContribute to development of standards and practices for Java, JavaScript,

Uploading the cleaned dataset to the mongodb atlas server

In [15]:
import pymongo

'MongoClient' creates a connection between python and mongodb atlas server

In [16]:
client = pymongo.MongoClient("mongodb+srv://jenieabraham:zCtWNFXdXjNOQLFb@cluster0.i6hkwqn.mongodb.net/?retryWrites=true&w=majority")

Creating a database named 'salary database' and a collection/table named 'salary'.

In [17]:
db = client["salary_database"]
collection=db['salary']

Converting each record in the salary dataset to a dictionary

In [18]:
data=salary_df.to_dict(orient="records")
data

[{'job_designation': 'Data Architect II',
  'percentile25_salary': 42167,
  'median_salary': 50115,
  'percentile75_salary': 58800,
  'salary_type': 'base',
  'currency': 'EUR',
  'salary_duration': 'P1Y',
  'location': 'Dublin, Ireland',
  'last_reviewed_datetime': '2021-03-19T00:00:00Z',
  'weblink': 'https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/dublin',
  'job_desc': 'Contribute to an architecture roadmap covering transactional, operational and business intelligence.\r\nContribute to development of standards and practices for Java, JavaScript, and SQL development.\r\nContributes to product and solution designs to address critical issues and complex problems.\r\n',
  'county': 'Dublin',
  'country': ' Ireland',
  'last_reviewed_date': Timestamp('2021-03-19 00:00:00')},
 {'job_designation': 'Data Architect II',
  'percentile25_salary': 39972,
  'median_salary': 47506,
  'percentile75_salary': 55739,
  'salary_type': 'base',
  'currency': 'EUR',
  'sa

Uploading the data in the dictionary format to the MongoDB server

In [19]:
db.salary.insert_many(data)

Importing the data from the MongoDB server back to python. A new feature called '_id' has been generated

In [22]:
#Loading data from mongodb to python
salary_mongodb=db.salary
salary_mongodb = pd.DataFrame(list(salary_mongodb.find()))
salary_mongodb

_id    job_designation  percentile25_salary  \
0    639c6ca9af67407ac1664875  Data Architect II                42167   
1    639c6ca9af67407ac1664876  Data Architect II                39972   
2    639c6ca9af67407ac1664877  Data Architect II                39035   
3    639c6ca9af67407ac1664878  Data Architect II                39420   
4    639c6ca9af67407ac1664879  Data Architect II                38646   
..                        ...                ...                  ...   
121  639c6ca9af67407ac16648ee   Warehouse Helper                22562   
122  639c6ca9af67407ac16648ef   Warehouse Helper                22033   
123  639c6ca9af67407ac16648f0   Warehouse Helper                22251   
124  639c6ca9af67407ac16648f1   Warehouse Helper                21814   
125  639c6ca9af67407ac16648f2   Warehouse Helper                21406   

     median_salary  percentile75_salary salary_type currency salary_duration  \
0            50115                58800        base      EUR             P1Y   
1            47506                55739        base      EUR             P1Y   
2            46393                54433        base      EUR             P1Y   
3            46850                54969        base      EUR             P1Y   
4            45930                53890        base      EUR             P1Y   
..             ...                  ...         ...      ...             ...   
121          29382                38055        base      EUR             P1Y   
122          28693                37163        base      EUR             P1Y   
123          28976                37530        base      EUR             P1Y   
124          28407                36793        base      EUR             P1Y   
125          27877                36106        base      EUR             P1Y   

                  location last_reviewed_datetime  \
0          Dublin, Ireland   2021-03-19T00:00:00Z   
1            Cork, Ireland   2021-03-19T00:00:00Z   
2          Galway, Ireland   2021-03-19T00:00:00Z   
3        Limerick, Ireland   2021-03-19T00:00:00Z   
4    Shannon-West, Ireland   2021-03-19T00:00:00Z   
..                     ...                    ...   
121          Cork, Ireland   2021-03-19T00:00:00Z   
122        Galway, Ireland   2021-03-19T00:00:00Z   
123      Limerick, Ireland   2021-03-19T00:00:00Z   
124  Shannon-West, Ireland   2021-03-19T00:00:00Z   
125     Waterford, Ireland   2021-03-19T00:00:00Z   

                                                                                          weblink  \
0          https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/dublin   
1            https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/cork   
2          https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/galway   
3        https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/limerick   
4    https://www.salary.com/research/ie-salary/benchmark/data-architect-ii-salary/ie/shannon-west   
..                                                                                            ...   
121           https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/cork   
122         https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/galway   
123       https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/limerick   
124   https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/shannon-west   
125      https://www.salary.com/research/ie-salary/alternate/warehouse-helper-salary/ie/waterford   

                                                                                                                                                                                                                                                                                                                                                  

In [21]:
salary_mongodb.shape
#An additional column 'id' has been introduced

(126, 15)